SQL_DDL_Func_HW

crisforp opened the pull request:
Tasks: applying view and functions: 1 - 5

crisforp committed:
Add files via upload

Tasks: applying view and functions: 1 - 5

mozuela commented:
Hi Cristian, 
Task 1:
Join logic correct, properly dropping the view, and filtering 
CEIL is fine, just keep in mind PostgreSQL has QUARTER built-in 
Category_name column name is duplicate, looks like a typo

Task 2:
Well use of raise exception and null checks, good detail casting money. very good job with the testing scenarios and comments along the code. 
1) input parameter should be just one: date instead of qtr + yr, please update it

Task 3: 
Nice implementation of CTE, trim() usage and comments 
1) upper is missed for Country 
2) input parameter should be an array/multiple countries, no just one 
3) Ensure not duplicates 

Task 4: 
I really like the use of ILIKE instead of upper, well implementation of raise exception and very well code documentation
1) For row_number window function shouldn't be use, please implemente different 

Task 5: 
1) UPPER() missed for language 
2) make sure not duplicates will be insert, movies with the same title 
Well used of the defaults and checking if language exists and raise exception

crisforp committed:
Add files via upload

Hello!

I modified the code according to my mentor's requirements
Task 1:
Join logic correct, properly dropping the view, and filtering
CEIL is fine, just keep in mind PostgreSQL has QUARTER built-in
Category_name column name is duplicate, looks like a typo

Task 2:
Well use of raise exception and null checks, good detail casting money. very good job with the testing scenarios and comments along the code.

input parameter should be just one: date instead of qtr + yr, please update it
Task 3:
Nice implementation of CTE, trim() usage and comments

upper is missed for Country
input parameter should be an array/multiple countries, no just one
Ensure not duplicates
Task 4:
I really like the use of ILIKE instead of upper, well implementation of raise exception and very well code documentation

For row_number window function shouldn't be use, please implemente different
Task 5:

UPPER() missed for language
make sure not duplicates will be insert, movies with the same title
Well used of the defaults and checking if language exists and raise exception

crisforp committed:
Add files via upload

Task 1 - 5 updated
Task 6 added (partially)

mozuela commented:
b) function display 0 because is using previous 3 months, however data is older than that, try to update the code to consider it. 
d) In case customer is overdue 2x the rental duration, the replacement cost should be charge, try to update the function to include that

crisforp committed:
Add files via upload

Updates solution for Task 6 bases on my mentor's comments:
b) function display 0 because is using previous 3 months, however data is older than that, try to update the code to consider it.
d) In case customer is overdue 2x the rental duration, the replacement cost should be charge, try to update the function to include that
